### PR TITLE
Compile track.c as C++, fix preview Z offsets

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/ride/Track.cpp [hunk 1]
```diff
@@ -345,8 +347,8 @@ const rct_trackdefinition FlatRideTrackDefinitions[256] = {
     { TRACK_FLAT,                   TRACK_SLOPE_DOWN_25,        TRACK_SLOPE_DOWN_25,        TRACK_BANK_NONE,        TRACK_BANK_NONE,        0                           },  // 37
     { TRACK_S_BEND,                 TRACK_SLOPE_NONE,           TRACK_SLOPE_NONE,           TRACK_BANK_NONE,        TRACK_BANK_NONE,        0                           },  // 38
     { TRACK_S_BEND,                 TRACK_SLOPE_NONE,           TRACK_SLOPE_NONE,           TRACK_BANK_NONE,        TRACK_BANK_NONE,        0                           },  // 39
-    { TRACK_VERTICAL_LOOP,          TRACK_SLOPE_DOWN_25,        TRACK_SLOPE_UP_25,          TRACK_BANK_NONE,        TRACK_BANK_NONE,        TRACK_UNKNOWN_VERTICAL_LOOP },  // 40
-    { TRACK_VERTICAL_LOOP,          TRACK_SLOPE_DOWN_25,        TRACK_SLOPE_UP_25,          TRACK_BANK_NONE,        TRACK_BANK_NONE,        TRACK_UNKNOWN_VERTICAL_LOOP },  // 41
+    { TRACK_VERTICAL_LOOP,          TRACK_SLOPE_DOWN_25,        TRACK_SLOPE_UP_25,          TRACK_BANK_NONE,        TRACK_BANK_NONE,        -48                         },  // 40
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: are you really introducing unnamed numbers?

**Comment by janisozaur** [comment]
I wish you kept C++ and formatting changes separate. Can you amend the history to split them?

File: src/openrct2/ride/Track.cpp [hunk 2]
```diff
@@ -35,276 +36,277 @@
 #include "ride_ratings.h"
 #include "RideGroupManager.h"
 #include "station.h"
-#include "track.h"
+#include "Track.h"
 #include "track_data.h"
-#include "../world/map.h"
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: Includes should be sorted

File: src/openrct2/ride/Track.cpp [hunk 3]
```diff
@@ -654,26 +666,30 @@ static bool track_add_station_element(sint32 x, sint32 y, sint32 z, sint32 direc
             ride->station_starts[stationIndex].x = (x >> 5);
             ride->station_starts[stationIndex].y = (y >> 5);
             ride->station_heights[stationIndex] = z;
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: misaligned with assignment above

File: src/openrct2/ride/Track.cpp [hunk 4]
```diff
@@ -734,27 +761,33 @@ static bool track_add_station_element(sint32 x, sint32 y, sint32 z, sint32 direc
                     ride->station_starts[stationIndex].x = (x >> 5);
                     ride->station_starts[stationIndex].y = (y >> 5);
                     ride->station_heights[stationIndex] = z;
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: misaligned with assignment above

File: src/openrct2/ride/Track.cpp [hunk 5]
```diff
@@ -1406,12 +1563,14 @@ static money32 track_place(sint32 rideIndex, sint32 type, sint32 originX, sint32
              TrackPricing[type];
 
     price >>= 16;
-    price = cost + ((price / 2) * 10);
+    price         = cost + ((price / 2) * 10);
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: ?

File: src/openrct2/ride/Track.cpp [hunk 6]
```diff
@@ -1679,20 +1868,23 @@ static money32 track_remove(uint8 type, uint8 sequence, sint16 originX, sint16 o
...
         price *= FlatRideTrackPricing[type];
     }
-    else {
+    else
+    {
         price *= TrackPricing[type];
     }
     price >>= 16;
-    price = (price + cost) / 2;
+    price         = (price + cost) / 2;
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: ?

File: src/openrct2/ride/Track.cpp [hunk 7]
```diff
@@ -1992,35 +2223,37 @@ void game_command_set_maze_track(sint32 *eax, sint32 *ebx, sint32 *ecx, sint32 *
  *
  *  rct2: 0x006C5AE9
  */
-void game_command_set_brakes_speed(sint32 *eax, sint32 *ebx, sint32 *ecx, sint32 *edx, sint32 *esi, sint32 *edi, sint32 *ebp)
+void game_command_set_brakes_speed(sint32 * eax, sint32 * ebx, sint32 * ecx, sint32 * edx, sint32 * esi, sint32 * edi, sint32 * ebp)
 {
-    rct_map_element *mapElement;
-    sint32 x, y, z, trackType, brakesSpeed;
+    rct_map_element * mapElement;
+    sint32          x, y, z, trackType, brakesSpeed;
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: alignment should match variable names, not type specifiers, i.e.
```
rct_map_element * mapElement;
sint32            x;
```
not 
```
rct_map_element * mapElement;
sint32          x;
```

File: src/openrct2/ride/Track.cpp [hunk 8]
```diff
@@ -2194,40 +2450,46 @@ void track_element_set_inverted(rct_map_element * mapElement, bool inverted)
...
 {
-    uint8 colourThingToXor = (vehicle->update_flags >> 9) & 0xFF;
-    sint32 trackType = mapElement->properties.track.type;
-    sint32 rideType = get_ride(mapElement->properties.track.ride_index)->type;
-    sint32 trackColour = mapElement->properties.track.colour ^ colourThingToXor;
-    sint32 bankStart = TrackDefinitions[trackType].bank_start;
+    uint8  colourThingToXor = (vehicle->update_flags >> 9) & 0xFF;
+    sint32 trackType        = mapElement->properties.track.type;
+    sint32 rideType         = get_ride(mapElement->properties.track.ride_index)->type;
+    sint32 trackColour      = mapElement->properties.track.colour ^colourThingToXor;
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: space after `^`

File: src/openrct2/ride/Track.cpp [hunk 9]
```diff
@@ -888,77 +950,93 @@ static bool track_remove_station_element(sint32 x, sint32 y, sint32 z, sint32 di
...
             finaliseStationDone = false;
         }
-    } while (!finaliseStationDone);
+    }
+    while (!finaliseStationDone);
 
     return true;
 }
 
 static money32 track_place(sint32 rideIndex, sint32 type, sint32 originX, sint32 originY, sint32 originZ, sint32 direction, sint32 brakeSpeed, sint32 colour, sint32 seatRotation, sint32 liftHillAndAlternativeState, sint32 flags)
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: Overlong line, likely not the only one.

File: src/openrct2/ride/Track.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: It is. Haven't you looked a few lines above?!

File: src/openrct2/ride/Track.cpp (hunk 5 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: autoformatter

File: src/openrct2/ride/Track.cpp (hunk 7 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: I know, I already fixed a few. It's the autoformatter,

File: src/openrct2/ride/Track.cpp (hunk 2 above)
Comment (janisozaur) [reply src/openrct2/ride/Track.cpp]: All of them. `footpath.h`, `Track.h`, `RideGroupManager.h` are misplaced

File: src/openrct2/ride/Track.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: Have you even taken the effort to look what this field is? It's preview_z_offset. It IS a number.

**Comment by Gymnasiast** [comment]
> I wish you kept C++ and formatting changes separate. 

What on earth? You did exactly the same! What point is renaming to C++ without fixing formatting anyway?

**Comment by janisozaur** [comment]
> You did exactly the same!

Huh? I think you are mistaken.

https://github.com/OpenRCT2/OpenRCT2/pull/6442/commits
https://github.com/OpenRCT2/OpenRCT2/pull/6432/commits
https://github.com/OpenRCT2/OpenRCT2/pull/6417/commits

File: src/openrct2/ride/Track.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: How is Track.h misplaced?! Track should be before track_data.

Comment (janisozaur) [reply src/openrct2/ride/Track.cpp]: Oh, in case-insensitive sorting that might be true. Fair enough.

File: src/openrct2/ride/Track.cpp (hunk 1 above)
Comment (janisozaur) [reply src/openrct2/ride/Track.cpp]: Yes, I've seen, it's just that I found it a bit misleading that you of all people would introduce unnamed numbers and wanted to confirm

**Comment by Gymnasiast** [comment]
I have split the renaming, formatting and Z offset fixes.

File: src/openrct2/ride/Track.cpp (hunk 2 above)
Comment (janisozaur) [reply src/openrct2/ride/Track.cpp]: `footpath` still has to be moved

File: src/openrct2/ride/Track.cpp (hunk 6 above)
Comment (janisozaur) [reply src/openrct2/ride/Track.cpp]: needs fixing

File: src/openrct2/ride/Track.cpp [hunk 10]
```diff
@@ -2194,40 +2508,46 @@ void track_element_set_inverted(rct_map_element * mapElement, bool inverted)
     }
 }
 
-sint32 track_get_actual_bank(rct_map_element *mapElement, sint32 bank)
+sint32 track_get_actual_bank(rct_map_element * mapElement, sint32 bank)
 {
-    Ride *ride = get_ride(mapElement->properties.track.ride_index);
+    Ride   * ride      = get_ride(mapElement->properties.track.ride_index);
```
Comment (janisozaur) [inline src/openrct2/ride/Track.cpp]: improper alignment

**Comment by janisozaur** [comment]
Other than 4 or so still outstanding formatting issues LGTM

File: src/openrct2/ride/Track.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: Not sure why this pops up. I have fixed this and it appears aligned in the IDE.

File: src/openrct2/ride/Track.cpp (hunk 6 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: Fixed, but this does not disappear for some reason.

File: src/openrct2/ride/Track.cpp (hunk 10 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: Now fixed.

File: src/openrct2/ride/Track.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Track.cpp]: Has been fixed.

**Pull request merged**